test/metrics/query.go

```diff
@@ -0,0 +1,260 @@
+package metrics
+
+// package metrics provides the following exported functions:
+//     LocatePrometheus
+//     CreateAnnotatedCABundleInjectionConfigMap
+//     PodExecCreate
+//     RunQueryFromPod
+// These functions can validate prometheus metrics
+// and also will validate the current cluster CA
+// issued through service-ca-operator.
+// To only validate the CA, one can pass the watchdog alert to RunQueryFromPod,
+// as this alert should always be firing in a healthy cluster:
+//     `ALERTS{alertstate="firing",alertname="Watchdog",severity="none"} == 1`
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"os/exec"
+	"strings"
+	"syscall"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	uexec "k8s.io/utils/exec"
+
+	"github.com/prometheus/common/model"
+)
+
+const (
+	prometheusBaseURL         = "https://prometheus-k8s.openshift-monitoring.svc:9091"
+	serviceCAInjectAnnotation = "service.beta.openshift.io/inject-cabundle"
+)
+
+func LocatePrometheus(client *kubernetes.Clientset) (string, error) {
```

Comment:
Missing godoc, what are the return values.

Reply:
updated

Reply:
updated